packages/resolver.go

```diff
@@ -0,0 +1,13 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package packages
+
+import "net/http"
+
+type RemoteResolver interface {
+	RedirectArtifactsHandler(w http.ResponseWriter, r *http.Request, p *Package)
+	RedirectStaticHandler(w http.ResponseWriter, r *http.Request, p *Package, resourcePath string)
+	RedirectSignaturesHandler(w http.ResponseWriter, r *http.Request, p *Package)
+}
```

Comment:
Nit. Actually, looking at this interface, this is also implemented by `ServePackage`, `ServeFile` and `ServeSignature`. Maybe all packages could have a "Handler", and each indexer provides an implementation, the local ones would use the `Serve*` methods, and the remote ones would use the specific remote resolver implementations.
If we do this, it could be in a future refactor.

Reply:
I would leave it for further refactoring. I think that this idea is similar to the first approach I proposed (packages hijack request handling).